### PR TITLE
Harden addSite and moveSite

### LINKED ISSUE #7014: Bookmarking into folders dupes or doesn't work
- Did you search for similar issues before submitting this one?
Yes

- Describe the issue you encountered:
Bookmarking into folders dupes or doesn't work.

- Platform (Win7, 8, 10? macOS? Linux distro?):
MacOS 10.11.6

- Brave Version (revision SHA):
0.13.2-rc1; master @cfa3186ddc7a0631044f482fe1758d72fb84ad83

- Steps to reproduce:
    0. Open Brave and enable the bookmarks toolbar.
    1. Create a folder on the toolbar.
    2. Visit a page. Bookmark it by clicking the URL bar star, specifying the folder you created.
    3. Bookmark is duplicated to be both in the folder and on the toolbar.
    4. Visit another page. Bookmark it by dragging the padlock onto the toolbar folder.
    5. Bookmark is not in the folder; instead it's on the toolbar.
    
- Actual result:
The bookmarking common case (Click the Star / Cmd+D) duplicates when bookmarking into folders.
Drag to bookmark can't bookmark into folders.

- Expected result:
Bookmarking into folders should work normally.

- Will the steps above reproduce in a fresh profile? If not what other info can be added?
Fresh– yes

- Is this an issue in the currently released version?
N (current RC)

- Can this issue be consistently reproduced?
Y

- Any related issues:
maybe #6510

### LINKED ISSUE #7016: Adding, removing, then re-adding a bookmark errors
- Did you search for similar issues before submitting this one?
Yes

- Describe the issue you encountered:
See title

- Platform (Win7, 8, 10? macOS? Linux distro?):
All

- Brave Version (revision SHA):
0.13.2-rc1

- Steps to reproduce:
    1. Visit brave.com
    2. Click the star button to add the bookmark
    3. Click the star button again and click on remove
    4.  Click the star button again
    
- Actual result:
The bookmark is not re-added and Shift+f8 console has an error

- Expected result:
No error and it re-adds the bookmark

- Will the steps above reproduce in a fresh profile? If not what other info can be added?
Yes

- Is this an issue in the currently released version?
No

- Can this issue be consistently reproduced?
Yes

- Screenshot if needed:
screenshot 2017-02-02 22 43 39

- Any related issues:

### PR DESCRIPTION
**Pull request by darkdh**
fix #7014
fix #7016

Auditors: @bbondy, @ayumi

Test Plan: covered by automatic test

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).

### CODE REVIEW

File: js/components/navigationBar.js [hunk 1]
```diff
@@ -91,11 +91,7 @@ class NavigationBar extends ImmutableComponent {
 
   get bookmarked () {
     return this.props.activeFrameKey !== undefined &&
-      siteUtil.isSiteBookmarked(this.props.sites, Immutable.fromJS({
-        location: this.props.location,
-        partitionNumber: this.props.partitionNumber,
-        title: this.props.title
-      }))
+      siteUtil.isSiteBookmarked(this.props.sites, Immutable.fromJS({location: this.props.location}))
```
Comment (bbondy) [inline js/components/navigationBar.js]: we differentiate on bookmarks with different partitions, at least until today with previous releases.  For example a user might want to bookmark twitter.com when logged in as user A and twitter.com when logged in as user B.

File: js/state/siteUtil.js [hunk 2]
```diff
@@ -75,19 +75,24 @@ module.exports.getSiteKey = function (siteDetail) {
...
   if (!sites) {
     return false
   }
-  const key = module.exports.getSiteKey(siteDetail)
-  if (key === null) {
-    return false
+
+  const site = sites.find((site) =>
+    isBookmark(site.get('tags')) &&
+    site.get('location') === siteDetail.get('location')
```
Comment (bbondy) [inline js/state/siteUtil.js]: ditto lack of partition here

Comment (darkdh) [inline js/state/siteUtil.js]: updated in f9861f9a72359cf5ea5af829e3b273151ef60d2c

**Comment by bbondy** [comment]
++

**Comment by luixxiul** [comment]
Will this be included in 0.13.2?

**Comment by darkdh** [comment]
Yes, it will be  in RC2

**Comment by ayumi** [comment]
@darkdh thanks! I can bookmark into folders now. :)

**Pull request merged**